CoreUtilities/src/au/gov/asd/tac/constellation/utilities/svg/SVGAttributeConstant.java

```diff
@@ -0,0 +1,100 @@
+/*
+* Copyright 2010-2023 Australian Signals Directorate
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package au.gov.asd.tac.constellation.utilities.svg;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * This class serves as a constant to fill the attribute key for SVG attributes.
+ * 
+ * @author capricornunicorn123
+ */
+public enum SVGAttributeConstant {
+    BASELINE("dominant-baseline"),
+    CLASS("class"),
+    CUSTOM_SORT_ORDER("data-sort-order"),
+    CX("cx"),
+    CY("cy"),
+    DESTINATION_X("x2"),
+    DESTINATION_Y("y2"),
+    DASH_ARRAY("stroke-dasharray"),
+    EXTERNAL_RESOURCE_REFERENCE("xlink:href"),
+    FILL_COLOR("fill"),
+    FILTER("filter"),
+    FONT_SIZE("font-size"),
+    HEIGHT("height"),
+    ID("id"),
+    NAME_SPACE("xmlns"),
+    POINTS("points"),
+    RADIUS("r"),
+    SOURCE_X("x1"),
+    SOURCE_Y("y1"),
+    STROKE_COLOR("stroke"),
+    TRANSFORM("transform"),
+    VIEW_BOX("viewBox"),
+    WIDTH("width"),
+    X("x"),
+    Y("y");
+    
+    private final String attributeName;
+    
+    private SVGAttributeConstant(final String attributeName) {
+        this.attributeName = attributeName;
+    }
+    
+    /**
+     * Returns the key value used by SVG attributes.
+     * @return 
+     */
+    public String getName() {
+        return this.attributeName;
+    }
+    
+    /**
+     * Generates a LinkedHashMap of important attribute values.
+     * Created for the SVGData constructor to ensure that significant attributes such as id, x and y are forced to the front of the svg tag. 
+     * @return 
+     */
+    public static Map<String, String> initialiseBasicAttributes() {
```

Comment:
just looking at the usage of this function, I'm not sure if it is entirely needed as when you retrieve values from a map, it will return null anyway if the key doesn't exist (and if you populate value with other values, that doesn't need an explicit initialise first)

Reply:
The LinkedHashMap that stores attributes ensures that element addition order is maintained with the intention of allowing the resulting svg string to have a predictable order of attributes. This has the side effect of not allowing developers to add attributes willy-nilly as the order of addition will effect the order of output - example of ad hoc element addition on output.

- <svg x="4" viewBox="0, 0, 32, 32" width="8" fill="$fff000" id="myID" y="7" />

As adding attributes in an ad hoc manner is convenient for development it was useful to have important attributes such as identifying, positional and color information at the front of the output attributes and in a predictable order. 

This map initialization ensures that the order of these common element are predictable. When converting to a string, the SVGData object "cleans" the attributes and removes and key value pairs from the map that have a value of null.

Removing this method will not add or detract from the visual output of the plugin. Is it worth keeping this to aid in future development or sould it be removed?